server/chat-plugins/chat-monitor.ts

```diff
@@ -45,74 +46,155 @@ const EVASION_DETECTION_SUBSTITUTIONS: {[k: string]: string[]} = {
+					buf[key].push(word);
+				}
+			}
+			return JSON.stringify(buf);
+		}, {throttle: force ? 0 : WRITE_THROTTLE_TIME});
+	}
 
-function renderEntry(location: string, word: Chat.FilterWord, punishment: string) {
-	return `${location}\t${word.word}\t${punishment}\t${word.reason || ''}\t${word.hits}\t${word.replacement || ''}\t${word.publicReason || ''}\r\n`;
-}
+	add(filterWord: Partial<Chat.FilterWord> & {list: string, word: string}) {
+		if (!filterWord.hits) filterWord.hits = 0;
+		const punishment = Chat.monitors[filterWord.list].punishment;
+		if (!filterWord.regex) {
+			filterWord.regex = this.generateRegex(
+				filterWord.word,
+				punishment === 'EVASION',
+				punishment === 'SHORTENER',
+				!!filterWord.replacement,
+			);
+		}
 
-function saveFilters(force = false) {
-	FS(MONITOR_FILE).writeUpdate(() => {
-		let buf = 'Location\tWord\tPunishment\tReason\tTimes\r\n';
-		for (const key in Chat.monitors) {
-			buf += filterWords[key].map(
-				word => renderEntry(Chat.monitors[key].location, word, Chat.monitors[key].punishment)
-			).join('');
+		if (filterWords[filterWord.list].some(val => String(val.regex) === String(filterWord.regex))) {
+			throw new Chat.ErrorMessage(`${filterWord.word} is already added to the ${filterWord.list} list.`);
 		}
-		return buf;
-	}, {throttle: force ? 0 : WRITE_THROTTLE_TIME});
-}
-
-export function addFilter(filterWord: Partial<Chat.FilterWord> & {list: string, word: string}) {
-	if (!filterWord.hits) filterWord.hits = 0;
-	const punishment = Chat.monitors[filterWord.list].punishment;
-	if (!filterWord.regex) {
-		filterWord.regex = generateRegex(
-			filterWord.word,
-			punishment === 'EVASION',
-			punishment === 'SHORTENER',
-			!!filterWord.replacement,
-		);
-	}
 
-	if (filterWords[filterWord.list].some(val => String(val.regex) === String(filterWord.regex))) {
-		throw new Chat.ErrorMessage(`${filterWord.word} is already added to the ${filterWord.list} list.`);
+		filterWords[filterWord.list].push(filterWord as Chat.FilterWord);
+		this.save(true);
 	}
 
-	filterWords[filterWord.list].push(filterWord as Chat.FilterWord);
-	saveFilters(true);
-}
+	load() {
+		const legacy = FS(LEGACY_MONITOR_FILE);
+		if (legacy.existsSync()) {
+			return process.nextTick(() => {
+				this.loadLegacy();
+				legacy.renameSync(LEGACY_MONITOR_FILE + '.backup');
+				Monitor.notice(`Legacy chatfilter data loaded and renamed to a .backup file.`);
+			});
+		}
 
-export function generateRegex(word: string, isEvasion = false, isShortener = false, isReplacement = false) {
-	try {
-		if (isEvasion) {
-			return constructEvasionRegex(word);
-		} else {
-			return new RegExp((isShortener ? `\\b${word}` : word), (isReplacement ? 'igu' : 'iu'));
+		const data = JSON.parse(FS(MONITOR_FILE).readIfExistsSync() || "{}");
+		for (const k in data) {
+			filterWords[k] = [];
+			// previously, this checked to be sure the monitor existed in Chat.monitors and that there was
```

Comment:
/* */ yn

Reply:
I think a row of `//`s is fine; I don't think we have a use case for `/*` other than doc comments where they're `/**`.